packages/phoenix-ng/projects/phoenix-app/cypress/e2e/cms.test.ts

```diff
@@ -0,0 +1,190 @@
+describe('CMS', () => {
+  it('should match the whole page', () => {
+    cy.visit('/cms').then(() => {
+      cy.contains('This may take a while').should('not.be.visible');
+      // in order to make the detector appear which loads in 2 - 2.5 seconds
+      cy.wait(2500);
+      cy.document().toMatchImageSnapshot();
+    });
+  });
+
+  describe('Phoenix Menu', () => {
+    it('should have options to Save and Load state', () => {
+      cy.get('.phoenix-menu-item-right .item-settings').click();
+
+      cy.get('.phoenix-menu-item').contains('Save state').should('exist');
+      cy.get('.phoenix-menu-item').contains('Load state').should('exist');
+
+      cy.get('.phoenix-menu-item-right .item-config-backdrop').click();
+    });
+
+    it('should display correct options under event data dropdown', () => {
+      cy.get('.phoenix-menu-item-right > button').eq(0).click();
+
+      cy.get('[data-testid="dropdown"]').eq(1).focus();
+      cy.focused()
+        .parentsUntil('.phoenix-menu-item-left')
+        .contains('Event Data')
+        .should('be.visible');
+
+      cy.get('[data-testid="dropdown"]').eq(1).click();
+
+      cy.get('.phoenix-menu-item').contains('Tracks').should('exist');
+      cy.get('.phoenix-menu-item').contains('Jets').should('exist');
+      cy.get('.phoenix-menu-item').contains('CaloClusters').should('exist');
+      cy.get('.phoenix-menu-item').contains('MuonChambers').should('exist');
+
+      cy.get('[data-testid="dropdown"]').eq(1).click();
+    });
+
+    it('should hide event data', () => {
+      cy.get('app-phoenix-menu-item .mat-slide-toggle')
+        .eq(0)
```

Comment:
Does this not work for reaching a phoenix menu item? https://github.com/HSF/phoenix/pull/495#discussion_r957673738

The way it is now is still prone to errors since we are looking for any `.mat-slide-toggle` and getting it's first index and then making sure we have the `Event Data` option. This could break because the position of `.mat-slide-toggle` may change.

Please let me know if I misunderstand something here.

Reply:
The problem when using that is - `Event Data` toggle gets triggered as well when we use the toggle for `Detector`. So, I fixed that. Pushing the changes..now it should work as expected.

Reply:
That doesn't happen on my side. I used the following standalone test to check:

```ts
it('should hide detector geometry', () => {
  cy.get('[data-testid="dropdown"]').click();

  getPhoenixMenuItem('Detector').closest('.mat-slide-toggle').click();

  cy.document().toMatchImageSnapshot();

  getPhoenixMenuItem('Detector').closest('.mat-slide-toggle').click();

  cy.get('.phoenix-menu-item-right > button').eq(0).click();
});
```

This does not toggle "Event Data". Can you please share a screenshot how the "Event Data" gets triggered on clicking the "Detector" menu item?

Reply:
Now, I got it why that happened with this line just before the `hide detector geometry` test case - 
```
cy.get('app-phoenix-menu-item').eq(1).click();
```

And due to this, the `Event Data` toggle gets triggered.

Reply:
Does it look okay now or I need to change something else?  😅

Reply:
Is okay now.

Just note that the snapshots don't work on my Linux because of differences in rendering the canvas. There are subtle differences in how WebGL works on different operating systems. I am not sure if it will work on some other Windows machine either. I will have to check. But just as a heads up, we may need to have a docker setup to make things consistent.

Reply:
> we may need to have a docker setup to make things consistent.

For snapshots, definitely a yes! They have been inconsistent as per my check as well.